docs/minio-job.md

```diff
@@ -0,0 +1,210 @@
+# MinioJob is a Kubernetes Job that runs mc commands
+here is an example of a MinioJob:
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: mc-job-sa
+---
+apiVersion: sts.min.io/v1alpha1
+kind: PolicyBinding
+metadata:
+  name: mc-job-binding
+spec:
+  application:
+    serviceaccount: mc-job-sa
+  policies:
+    - consoleAdmin
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mytestsecret
+data:
+  PASSWORD: ZGFuaWVsMTIz
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mytestsecretenvs
+data:
+  USER: ZGFuaWVs
+  PASSWORD: ZGFuaWVsMTIz
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: mytestconfig
+data:
+  policy.json:  |
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "s3:*"
+                ],
+                "Resource": [
+                    "arn:aws:s3:::memes",
+                    "arn:aws:s3:::memes/*"
+                ]
+            }
+        ]
+    }
+---
+apiVersion: job.min.io/v1alpha1
+kind: MinIOJob
+metadata:
+  name: minio-test-job
+spec:
+  mcImage: minio/mc:latest
+  serviceAccountName: mc-job-sa
+  securityContext: {}
+  containerSecurityContext: {}
+  tenant:
+    name: mytest-minio
+  commands:
+    - op: make-bucket
+      args:
+        name: memes
+    - name: add-my-user-1
+      op: admin/user/add
+      args:
+        user: ${USER}
+        password: ${PASSWORD}
+      envFrom:
+        - secretRef:
+            name: mytestsecretenvs
+    - name: add-my-user-2
+      op: admin/user/add
+      args:
+        user: pedro
+        password: $PASSWORD
+      env:
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mytestsecret
+              key: PASSWORD
+    - name: add-my-policy
+      op: admin/policy/create
+      args:
+        name: memes-access
+        policy: /temp/policy.json
+      volumeMounts:
+        - name: policy
+          mountPath: /temp
+      volumes:
+        - name: policy
+          configMap:
+            name: mytestconfig
+            items:
+              - key: policy.json
+                path: policy.json
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: daniel
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: pedro
+    - op: stat
+      command:
+        - "mc"
+        - "stat"
+        - "myminio/memes"
+        - "--insecure"  
+```
+The MinioJob is a Kubernetes Job that runs mc commands. It uses the MinIO client (mc) to interact with the MinIO server.
+## mcImage
```

Comment:
I guess we should document it (we can refer to standard K8s documentation for these fields).

Reply:
> Yes, the standard attribute `image` does look better, but if you don't remind that this is a image of `mc`, I believe many people will use the minio image to deal with it, because it is `latest` by default

I don't really mind that we're using `mcImage` instead of `image` (although most users would leave it at the default value), so I would write in the documentation that the default should be fine and you only specify your own image if you absolutely need to. I would comment this item in this example, because we don't want users to specify it.

If we specify one, then please use `quay.io/minio/mc:latest` instead, because that's the repository that we promote. 

We should also think about using `quay.io/minio/mc:latest` as a default, because that would not use a fixed `mc` version and doesn't make it repeatable (I prefer repeatable). The operator also doesn't use `latest` for Minio deployments, but also a fixed version. I think we should be consistent with that. On the other hand `mc` is quite stable, so maybe @harshavardhana should decide what to use.